corehq/apps/locations/views.py

```diff
@@ -179,20 +200,41 @@ def page_context(self):
             ))
         return {
             'form': self.location_form,
+            'products_per_location_form': self.products_form,
             'location': self.location,
             'consumption': consumption,
         }
 
-    def post(self, request, *args, **kwargs):
+    def form_valid(self):
+        messages.success(self.request, _('Location saved!'))
+        return HttpResponseRedirect(
+            reverse(self.urlname,
+                    args=[self.domain, self.location_form.location._id]),
+        )
+
+    def settings_form_post(self, request, *args, **kwargs):
         if self.location_form.is_valid():
             self.location_form.save()
-            messages.success(request, _('Location saved!'))
-            return HttpResponseRedirect('%s?%s' % (
-                reverse(LocationsListView.urlname, args=[self.domain]),
-                urllib.urlencode({'selected': self.location_form.location._id})
-            ))
+            return self.form_valid()
         return self.get(request, *args, **kwargs)
 
+    def products_form_post(self, request, *args, **kwargs):
+        products = SQLProduct.objects.filter(
+            product_id__in=request.POST.getlist('selected_ids', [])
+        )
+        self.sql_location.products = products
+        self.sql_location.save()
+        return self.form_valid()
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Is it possible for the user to make changes on one tab, then switch to the other and make more changes before finally saving? Will all changes be saved in that case?